### PR TITLE
ivar upgrade and add primer_scheme_bedfiles support

### PR DESCRIPTION
**Pull request by pvanheus**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by pvanheus** [comment]
1. upgrade ivar tools to ivar 1.2.1
2. add support in `ivar trim` for using scheme BED from `primer_scheme_bedfiles` tool-data table

**Comment by pvanheus** [comment]
@thanhleviet have a look if you have time.

**Comment by bgruening** [comment]
@pvanheus you could add a test for the location file.

**Comment by pvanheus** [comment]
> @pvanheus you could add a test for the location file.

test (#4 for `ivar trim`) added. Thanks to @bernt-matthias for help in getting it working.

File: tools/ivar/ivar_trim.xml [hunk 1]
```diff
@@ -40,15 +60,29 @@
...
-            <param name="input_bed" value="zika/db/zika_primers.bed" />
             <output name="output_bam" file="zika/Z52_b.trimmed.sorted.bam" compare="sim_size" delta="100000"/>
         </test>
+        <test>
+            <conditional name="primer">
+                <param name="source" value="cached" />
+                <param name="cached_bed" value="ARTICv1" />
+            </conditional>
+            <param name="input_bam" value="sars-cov-2/sars_cov2_untrimmed.bam" ftype="bam" />
+            <output name="output_bam" file="sars-cov-2/sars_cov2_trimmed.bam" compare="sim_size" />
```
Comment (bgruening) [inline tools/ivar/ivar_trim.xml:84]: BAM files can be compared without sim_size ... Galaxy will unpack them and compare the SAM files.

Comment (pvanheus) [reply tools/ivar/ivar_trim.xml]: Noted.

**Comment by bgruening** [comment]
Merci!

**Pull request merged**